src/workers/parser/mapExpression.js

```diff
@@ -30,22 +31,36 @@ export default function mapExpression(
     originalExpression: false
   };
 
+  let ast;
+  const getAst = () => {
+    if (ast) {
+      return ast;
+    }
+
+    try {
+      ast = parseConsoleScript(expression);
+    } catch (e) {
+      // no need to do anything
+    }
+    return ast;
+  };
```

Comment:
Why is this needed? Are we trying to prevent unnecessary parsing?

Reply:
Yes. we may end up doing 3 mappings, or none, so this should prevent doing multiple ones. Also, we don't want to bail out on the entire mapping since we do want to still map the await expression if it has a syntax error